cmake/FindZ3.cmake

```diff
@@ -1,29 +1,36 @@
 if (USE_Z3)
-    find_path(Z3_INCLUDE_DIR NAMES z3++.h PATH_SUFFIXES z3)
-    find_library(Z3_LIBRARY NAMES z3)
-    find_program(Z3_EXECUTABLE z3 PATH_SUFFIXES bin)
+    find_package(Z3 QUIET CONFIG)
```

Comment:
The ``CONFIG`` argument makes cmake ignore the current ``FindZ3.cmake`` file and only looks for the Z3 generated Z3 target configuration files (in ``${CMAKE_PREFIX_PATH}/lib/cmake/z3/*`` or something like that, if present).